### ISSUE TITLE
Should registry be a class method?

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#159]
When creating a new parameter/node/etc the API looks like this:

```
parameter_registry.add(ControlCurveIndexParameter)
```

How about this instead:

```
ControlCurveIndexParameter.register()
```

Presumably there would be an equivalent `deregister` or `unregister` function.

**Comment by jetuk** [comment]
What's the benefit of this? Just style?

**Comment by snorfalorpagus** [comment]
Yes.

**Comment by snorfalorpagus** [comment]
This was implemented in #265.

It could also be done for recorders...
